scripts/trace_flowcell.py

```diff
@@ -0,0 +1,152 @@
+import os
+import requests
+import json
+import sys
+import subprocess
+import yaml
+import argparse
+from datetime import datetime
+
+class GetTraces:
+    
+    def __init__(self, api_base_url, access_headers, req_verify, workflow):
+        self.api_base_url = api_base_url
+        self.headers = access_headers
+        self.verify = req_verify
+        self.workflow = workflow
+    
+    def get_executions_for_tag(self, tag):
+        traces_response = requests.get(
+            "{}/{}/?{}={}".format(self.api_base_url, "traces", "trace_tag", tag),
+            verify = self.verify,
+            headers = self.headers)
+        if not traces_response.ok:
+            raise Exception("Response not OK, got: " + traces_response.text)
+        traces = json.loads(traces_response.text)
+        for trace in traces:
+            trace_id = trace["id"]
+            trace_info_response = requests.get(
+                "{}/{}/{}".format(self.api_base_url, "traces", trace_id),
+                verify = self.verify,
+                headers = self.headers)
+            trace_info = json.loads(trace_info_response.text)
+            action_executions = map(lambda x: x["object_id"], trace_info["action_executions"])
+            for execution in action_executions:
+                yield execution
+
+    def get_actions_from_executions(self, executions):
+        for execution_id in executions:
+            execution_info_response = requests.get(
+                "{}/{}/{}".format(self.api_base_url, "executions", execution_id),
+                verify = self.verify,
+                headers = self.headers)
+            execution_info = json.loads(execution_info_response.text)
+            yield execution_info
+    
+    def filter_and_order_actions_by_name(self, actions):
+        filtered_actions = (action for action in actions if self.workflow in action["action"]["name"])
+        def get_start_time(action):
+            start_time = datetime.strptime(action['start_timestamp'].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+            return start_time
+        sort_actions = sorted(filtered_actions, key = get_start_time)
+        return sort_actions
+
+    def get_sorted_actions_from_tag(self, tag):
+        executions = self.get_executions_for_tag(tag)
+        actions = self.get_actions_from_executions(executions)
+        return self.filter_and_order_actions_by_name(actions)
+
+class PickRunfolder:
+    
+    def __init__(self, config):
+        with open(config, 'r') as ymlfile:
+            cfg = yaml.load(ymlfile)
+        self.hosts = cfg['runfolder_svc_urls']
+
+    def load_folders(self):
+        for host in self.hosts:
+            url_base = '/'.join(host.split('/')[:-1])
+            result = requests.get("{}?state=*".format(url_base))
+            result_json = json.loads(result.text)
+            return result_json["runfolders"]
```

Comment:
if you were to keep to returning generators, you could just do a 
```suggestion
     for runfolder in result_json["runfolders"]:
         yield runfolder
```